From c2032b3e5b3f8260a92d48b9a340bb512e121fea Mon Sep 17 00:00:00 2001
From: jkreierh <jkreierh@90b65887-3827-0410-9a23-83215b262276>
Date: Mon, 30 Mar 2015 13:32:10 +0000
Subject: [PATCH 0578/1000] MANTIS 36057 Mantis 36057: conveyor: device handler
 should send baggage-present/baggage-absent for each belt in the status -
 Change "ConveyorDev.idl":   - to Version 1.4   - include new boolean
 attributes insertionBeltOccupied, verificationBeltOccupied,
 parkingBeltOccupied   - include a new complex attribute subDeviceStatusList
 (necessary for extended monitoring)

git-svn-id: svn://localhost/SelfServiceCommon/trunk@843 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/idl/Massai/ConveyorDev.idl         | 48 ++++++++++++++++++-
 1 file changed, 47 insertions(+), 1 deletion(-)

diff --git a/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl b/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl
index c2833f1d..bfa5c805 100644
--- a/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl
+++ b/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl
@@ -39,7 +39,7 @@
 **/
 module conveyordev
 {
-   const string interface_version = "1.3";
+   const string interface_version = "1.4";
 /*
 using DeviceComponents::indicationListener and DeviceComponents::Generic from mdci.idl
   RC indicationListener::indication(in any ind);
@@ -171,6 +171,29 @@ using DeviceComponents::indicationListener and DeviceComponents::Generic from md
                                          critical to health, safety and security */
    };
 
+   /**
+    *
+    * Definition of status of a specific device in the baggage handling system.
+    *
+   **/
+   enum subDeviceStatus
+   {
+      SD_OK,                        /**< Subdevice ok */
+      SD_DISCONNECTED,              /**< Subdevice is disconnected */
+      SD_HARDWARE_ERROR,            /**< Subdevice hardware error */
+      SD_HARDWARE_DEGRADED          /**< Subdevice warning "degraded" */
+   };
+
+
+   struct moduleStatus
+   {
+     moduleCharacteristic type;
+     subDeviceStatus status;
+   };
+
+   typedef sequence<moduleStatus> moduleStatusList;
+
+
    /**
     *
     * Defines the device characteristics.
@@ -225,6 +248,29 @@ using DeviceComponents::indicationListener and DeviceComponents::Generic from md
       **/
       errorType         errorStatus;
 
+      // The status of the sub devices.
+      moduleStatusList subDeviceStatusList;
+
+      /**
+       *
+       * Is there a bag on the insertion belt?
+       *
+      **/
+      boolean insertionBeltOccupied;
+
+      /**
+       *
+       * Is there a bag on the verification belt?
+       *
+      **/
+      boolean verificationBeltOccupied;
+
+      /**
+       *
+       * Is there a bag on the parking belt?
+       *
+      **/
+      boolean parkingBeltOccupied;
 
    };
 
-- 
2.41.0.windows.1

